src/typebox.ts

```diff
@@ -299,6 +305,14 @@ export class TypeBuilder {
             { ...options, kind: ObjectKind, type: 'object', additionalProperties, properties }
     }
 
+    /** `STANDARD` Creates a `$ref` schema. */
+    public Ref<T extends TObject<any>>(ref: T, options: CustomOptions = {}): TRef<T> {
+        if (!ref.$id) {
+            throw new Error('Referenced schema $id is missing')
```

Comment:
Ideally `Type.Ref` would require an `$id` defined for its argument.

The problem is that `CustomOptions` (or options in general) are not encoded into the type:
```ts
const Person = Type.Object({
    name: Type.String()
}, {
    $id: 'person'
})

// we should know that Person has an $id defined
const id: string = Person.$id // TS2322: Type 'string | undefined' is not assignable to type 'string'.
```

However that seems like a big change outside the scope of this PR